jetbrains-core/tst/software/aws/toolkits/jetbrains/services/s3/S3ServiceNodeTest.kt

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package software.aws.toolkits.jetbrains.services.s3
+
+import com.intellij.testFramework.ProjectRule
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.stub
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Rule
+import com.nhaarman.mockitokotlin2.any
+import org.junit.Test
+import software.amazon.awssdk.http.SdkHttpResponse
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.HeadBucketRequest
+import software.amazon.awssdk.services.s3.model.HeadBucketResponse
+import software.amazon.awssdk.services.s3.model.ListBucketsRequest
+import software.amazon.awssdk.services.s3.model.ListBucketsResponse
+import software.amazon.awssdk.services.s3.model.Bucket
+
+import software.aws.toolkits.jetbrains.core.MockClientManagerRule
+import software.aws.toolkits.jetbrains.utils.delegateMock
+import java.time.Instant
+
+class S3ServiceNodeTest {
+
+    @Rule
+    @JvmField
+    val projectRule = ProjectRule()
+
+    @JvmField
+    @Rule
+    val mockClientManagerRule = MockClientManagerRule(projectRule)
+
+    private val mockClient: S3Client by lazy { mockClientManagerRule.register(S3Client::class, delegateMock()) }
+
+    @Test
+    fun s3BucketsSortedAlphabetically() {
+
+        val mockSdkResponse = mock<SdkHttpResponse>()
+        mockSdkResponse.stub {
+            on { headers() } doReturn (mapOf("x-amz-bucket-region" to listOf("us-east-1")))
```

Comment:
I think I prefer the infix invocation of `doReturn` to remove extra parenthesis.

```kotlin
on { headers() } doReturn mapOf("x-amz-bucket-region" to listOf("us-east-1"))
```

@abrooksv There are too many ways of using mockito-kotlin. What should we standardize on?

```kotlin
`when`(something).thenReturn(something)
whenever(something).thenReturn(something)
on { something }.thenReturn(something)
on { something } doReturn something
on { something }.doReturn(something)
...
...
a bunch of other ways of doing the same thing
...
...
```

Reply:
I do not have a strong preference, but I agree we should standardize. Thanks mockito for all the aliases...